department-of-veterans-affairs/abd-vro
# 1591 implement integration test for api gateway

## What was the problem?


Associated tickets or Slack threads:
- #1591 

## How does this fix it?[^1]

Integration test hits endpoints of swagger docs for abd-vro, vro-app, and cc-app

## How to test this PR
- Step 1 - Follow steps to start platform base and API Gateway [here](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base--api-gateway)
- Step 2 - Run cc-app:  
```
./gradlew domain-cc:cc-app:dockerStart
```
- Step 3 - Run vro-app:
```
./gradlew :app:dockerComposeUp
```
- Step 4 - Follow steps for testing API Gateway [here](https://github.com/department-of-veterans-affairs/abd-vro/wiki/API-Gateway#testing-the-api-gateway-locally)
- Step 5 - Change the dropdown in the top right to select each of the three domains and verify successful load of API docs.
- Step 6 - See successful integration test run [here](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/5469376784/jobs/9958242763)


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
